veggieseasons/lib/screens/settings.dart

```diff
@@ -149,12 +155,7 @@ class SettingsScreen extends StatelessWidget {
         },
       ),
       onPress: () {
-        Navigator.of(context).push<void>(
-          CupertinoPageRoute(
-            builder: (context) => CalorieSettingsScreen(),
-            title: 'Calorie Target',
-          ),
-        );
+        Navigator.of(context).restorablePush<void>(_calorieSettingsScreen);
```

Comment:
could the .of happen to be implemented by a RestorableNavigator? Or it's intended for the user to choose whether each navigation change is restorable or not?

Reply:
There's no RestorableNavigator class. Just like any other built-in widget, the Navigator will attempt to restore its state if it has been configured with a restoration id.